src/mariadb_connection.c

```diff
@@ -296,6 +296,27 @@ MrdbConnection_Initialize(MrdbConnection *self,
     return -1;
   }
 
+  if (charset)
```

Comment:
To be one the safe side, we need to set the character set after the connection was established. If configuration file will contain a different character set, it will be overwritten. Same is valid for init_command.